src/util/specific.py

```diff
@@ -137,3 +138,66 @@ def reset_cowed_views(cr, xmlid, key=None):
         [key, module, name],
     )
     return set(sum(cr.fetchall(), ()))
+
+
+def translation2jsonb_unofficial_fields(cr):
+    """Convert all translated unofficial fields to jsonb.
+
+    This helper function is designed to run only in an on-premise Odoo instance
+    where any unofficial modules you use are installed, after going through the
+    upgrade to version 16.0.
+
+    It will find all those fields and convert translations from the
+    ``_ir_translation`` table into the new JSONB format.
+
+    Any fields not loaded will be skipped.
+    """
+    cr.execute(
+        """
+        SELECT ARRAY_AGG(name)
+        FROM ir_module_module
+        WHERE author LIKE 'Odoo S.A.'
```

Comment:
This is brittle. It's better if you fully restrict the fields you need to upgrade.